### PR TITLE
Add support for installing static builds on systems without usable internet connections.

### PR DESCRIPTION
**Pull request by Ferroin**
##### Summary

This works in three steps:

1. The user creates an offline install source directory using the new `--prepare-offline-install-source` option for the kickstart script.
2. The user exposes that directory to the target system (either by copying it, or by sharing it in some other way).
3. The user runs the `install.sh` script from that directory on the target system, with whatever other install options they want to use.

The offline install source directory consists of:

- The static install archives for the selected version of Netdata (we unconditionally download all architectures at the moment because it’s simpler).
- The checksum file for the static install archives (we verify the checksums as part of creating the offline install source).
- A copy of the kickstart script that was used to create the directory.
- A stub script called `install.sh` which invokes the copy of the kickstart script with the correct value for the new `--offline-install-source` option to make it use the contents of the directory for the install.
- A file called `channel` that stores the release channel (used to set it automatically during the install).

The required changes in the actual installation code are relatively trivial, simply involving modifying `download` and `check_for_remote_file` to accept `file:///` URLs and `set_static_archive_urls` to generate such URLs when an offline install source has been specified.

##### Test Plan

Testing follows the general outline above. The `--prepare-offline-install-source` option takes exactly one argument, specifying the name of the directory to use (it will be created if it does not exist). Creation of the offline install source directory is expected to work on any platform with a working POSIX compliant shell, `wget` or `curl`, and `sha256sum` or `shasum`. Actual installation is expected to work on any platform which we provide static builds for (currently Linux systems running on 64-bit x86, ARMv7l, AArch64, or POWER8+).

##### Additional Information

Fixes: https://github.com/netdata/product/issues/2314
Fixes: #12111
Supersedes: https://github.com/netdata/netdata/pull/12802

Implementing this for local builds is theoretically possible, but would require significant changes to `netdata-installer.sh`.

<details> <summary>For users: How does this change affect me?</summary>
This enables support for installation of static builds on systems which do not have a usable internet connection.
</details>

### CODE REVIEW

**Comment by vkalintiris** [comment]
Why do we need this? Building from source is not the preferred installation process.

**Comment by Ferroin** [comment]
> Why do we need this? Building from source is not the preferred installation process.

This is for static installs at the moment, not building from source (though if there’s enough demand it should be possible to get it working when building from source). The use of ‘source’ in multiple places here is in the sense of ‘this is where things are coming from’, not the sense of ‘source code’.

**Comment by maneamarius** [comment]
There's an issue with sha256sum command on centos78. On centos8 and ubuntu21.10 it works fine
image

**Comment by iigorkarpov** [comment]
Any requirements to the platform where kickstart.sh runs?

**Comment by maneamarius** [comment]
Same issue with ubuntu 16.04.

**Comment by Ferroin** [comment]
> Any requirements to the platform where kickstart.sh runs?

Not really, just needs a POSIX compliant shell, either wget or cURL, and either sha256sum or shasum command.

The actual install will only work for systems that we support static installs on though (but even the CPU architecture does not need to match the system that produced the directory).

**Comment by Ferroin** [comment]
@maneamarius Should be fixed now.

**Review by maneamarius — APPROVED** [review]

**Review by iigorkarpov — APPROVED** [review]
Tested under

- Centos 7
- Rocky 8
- Ubuntu1 8
- Ubuntu 20
- Ubuntu 22
- Debian 10
- Debian 11
- Oracle Linux 8

**Comment by Ferroin** [comment]
Holding off on merging this until we get an approval on the docs from @DShreve2, and ideally until after https://github.com/netdata/netdata/pull/12729 is merged (special testing will be needed to verify that these changes interact properly with that PR, and I’d rather worry about making any fixes here than requiring @maneamarius to make them there).

**Review by maneamarius — APPROVED** [review]

File: packaging/installer/kickstart.sh
Comment (Dimitris-P) [inline packaging/installer/kickstart.sh:167]: > insallling

Typo here

File: packaging/installer/kickstart.sh
Comment (Dim-P) [inline packaging/installer/kickstart.sh:1763]: > ofline 

Another one!

**Review by Dim-P — APPROVED** [review]
Tested on Ubuntu 20.04 and Raspbian 11 and it works fine after the latest changes.

**Pull request merged**
